dask/array/slicing.py

```diff
@@ -428,13 +454,23 @@ def partition_by_size(sizes, seq):
     >>> partition_by_size([10, 20, 10], [1, 5, 9, 12, 29, 35])
     [[1, 5, 9], [2, 19], [5]]
     """
-    seq = np.array(seq)
-    right = np.cumsum(sizes)
-    locations = np.searchsorted(seq, right)
-    locations = [0] + locations.tolist()
-    left = [0] + right.tolist()
-    return [(seq[locations[i]:locations[i + 1]] - left[i]).tolist()
-            for i in range(len(locations) - 1)]
+    # seq = np.array(seq)
+    # right = np.cumsum(sizes)
+    # locations = np.searchsorted(seq, right)
+    # locations = [0] + locations.tolist()
+    # left = [0] + right.tolist()
+    # return [(seq[locations[i]:locations[i + 1]] - left[i]).tolist()
+    #         for i in range(len(locations) - 1)]
+
+    seq = np.asanyarray(seq)
+    left = np.empty(len(sizes) + 1, dtype=int)
+    left[0] = 0
+
+    right = np.cumsum(sizes, out=left[1:])
```

Comment:
good point. `np.searchsorted` doesn't have an `out` parameter, but it's still faster:
```ipython
In [18]: %timeit locations = [0] + np.searchsorted(seq, right).tolist()
100 loops, best of 3: 6.71 ms per loop

In [19]: %timeit locations = np.empty(len(sizes) + 1, dtype=int); locations[0] = 0; locations[1:] = np.searchsorted(seq, right)
100 loops, best of 3: 6.1 ms per loop
```